### PR TITLE
DrivAerML: 4L/256d+T_max=30 — dropout=0.1 regularization

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The baseline 4L/256d model (12.70%) is still converging at the epoch cap, which may indicate it is memorizing training geometry rather than learning generalizable pressure features. Adding dropout=0.1 to the transformer layers can act as implicit regularization, reducing overfitting and potentially improving generalization on the 500-car DrivAerML test set. Light dropout (0.1) is unlikely to hurt training convergence significantly but may smooth the learned surface representations.

## Instructions

Run the following command:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --dropout 0.1 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-4L256d-training-tricks
```

Key changes from baseline:
- `--dropout 0.1` added (check train.py for the exact flag name — may be `--dropout`, `--drop`, `--attn-dropout`, etc.)
- `--wandb_group drivaerml-4L256d-training-tricks` for grouping

All OOM-prevention flags are preserved: `--batch-size 1`, `--drivaerml-train-surface-points 50000`, `--drivaerml-eval-surface-points 50000`, `--max-train-batches 394`, `--max-eval-batches 200`, `--model-heads 4`.

Report val/loss and all surface MAE metrics (p_in, p_oodc, p_tan, p_re) and the primary metric `val_primary/surface_rel_l2_pct`. Compare against baseline of **12.70%**.

## Baseline

Current best: PR #2593 (shinji), W&B run: `3aaevlho`
- val_primary/surface_rel_l2_pct: **12.70%**
- Model: 4L/256d, T_max=30, AdamW lr=5e-4, no-EMA, Fourier features, 45 epochs (still converging)

Reproduce baseline:
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2650 because ADVISOR: Closing — DrivAerML baseline shifted to 5.73% (4L/384d, PR #2602). All 4L/256d experiments are now obsolete — the best 4L/256d result was 11.97%. Width scaling is the dominant lever. Redirecting your GPU to 4L/384d experiments. You'll receive a new assignment shortly..
